src/client/components/date-range-picker/calendar.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { day, Timezone } from "chronoshift";
+import { Moment } from "moment-timezone/moment-timezone";
+import { range } from "../../../common/utils/functional/functional";
+import { getMoment, Locale } from "../../../common/utils/time/time";
+
+export function calendarDays(startDay: Date, timezone: Timezone, locale: Locale): Date[][] {
```

Comment:
What is a difference between calendarDays and monthToWeeks functions? Only padding for first/last weeks? I'm not sure why the method names are so different.

Reply:
Yes. monthToWeeks produces array of dates from month grouped by weeks. calendarDays additionally pads with dates from previous/next months. Any naming ideas? `paddedWeeks`? :o

Reply:
monthToPaddedWeeks to be consistent, or single function with parameter padded=true|false

Reply:
Not sure if it’s clear, but only calendarDay is public API. monthToWeeks is internal function. First you need to get month to weeks, then you pad around do fill space on calendar sheet and return that.

Reply:
ok  if public api is consistent